src/catch2/reporters/catch_reporter_helpers.cpp

```diff
@@ -235,4 +235,88 @@ namespace Catch {
         out << "\n\n" << std::flush;
     }
 
+    namespace {
+        struct SummaryColumn {
+            SummaryColumn( std::string _label, Colour::Code _colour ):
+                label( CATCH_MOVE( _label ) ), colour( _colour ) {}
+            SummaryColumn addRow( std::uint64_t count ) {
```

Comment:
This is another big mess. I don't feel like saddling you with fixing the implementation, but from looking at it, the signature does not make sense either.

It returns a copy, but not before modifying the current state of the instance, so something like

`SummaryColumn&& addRow(std::uint64_t count) &&`

makes more sense IMO, so we do not copy all the rows every time (which makes repeated calls to `addRow` have quadratic complexity), and the `&&` qualifier means that this won't be modifying any lvalues.